source/Octopus.Cli/Util/CommandSuggester.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Packaging;
+using Octopus.Cli.Commands;
+
+namespace Octopus.Cli.Util
+{
+   public static class CommandSuggester
+    {
+        public static IEnumerable<string> SuggestCommandsFor(
+            string[] words,
+            IReadOnlyDictionary<string, string[]> completionItems)
+        {
+            // some shells will pass the command name as invoked on the command line.
+            // If so, strip them from the beginning of the array
+            words = words
+                    .Take(2).Except(new[] {"octo", "complete"}, StringComparer.OrdinalIgnoreCase)
```

Comment:
Sure this doesn't need `octo.exe`?

Reply:
Have taken a different approach - powershell seems to consistently pass through `octo` or `octo.exe` on all platforms so I can just safely drop the first word.

Reply:
(which I've taken care of in the profile script. Therefore the will only apply on linux)